From 647011394abab7da261e827a3148c0c89467a6f8 Mon Sep 17 00:00:00 2001
From: DCCooper <1866858@gmail.com>
Date: Wed, 10 Feb 2021 10:16:17 +0800
Subject: [PATCH 2/2] enhancement: remove empty lines when showing image list

Signed-off-by: DCCooper <1866858@gmail.com>
---
 cmd/cli/images.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmd/cli/images.go b/cmd/cli/images.go
index bded3617..19777ce8 100644
--- a/cmd/cli/images.go
+++ b/cmd/cli/images.go
@@ -100,5 +100,6 @@ func formatAndPrint(images []*pb.ListResponse_ImageInfo) {
 	tabulate := gotabulate.Create(lines)
 	tabulate.SetHeaders(title)
 	tabulate.SetAlign("left")
+	tabulate.SetDenseMode()
 	fmt.Print(tabulate.Render("simple"))
 }
-- 
2.27.0

