From e85ac3e92535311cbed6a6c6b1e714b31c1fc980 Mon Sep 17 00:00:00 2001
From: xiadanni <xiadanni1@huawei.com>
Date: Fri, 20 Nov 2020 09:46:40 +0800
Subject: [PATCH] isula-build: mask /proc/pin_memory

Signed-off-by: xiadanni <xiadanni1@huawei.com>
---
 constant.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/constant.go b/constant.go
index 9c3f281..a0abfc9 100644
--- a/constant.go
+++ b/constant.go
@@ -97,6 +97,7 @@ var (
 		"/proc/sig_catch",
 		"/proc/kbox",
 		"/proc/oom_extend",
+		"/proc/pin_memory",
 		"/sys/firmware",
 		"/proc/cpuirqstat",
 		"/proc/memstat",
-- 
2.17.1

