From 9a212f5488b2e626eac23b2d0ab2dc6e288df5e7 Mon Sep 17 00:00:00 2001
From: lixiang <lixiang172@huawei.com>
Date: Fri, 27 Nov 2020 17:12:53 +0800
Subject: [PATCH] hack:add compile flag ftrapv

Signed-off-by: lixiang <lixiang172@huawei.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7e268678..f40941b4 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ BUILDTAGS := seccomp
 BUILDFLAGS := -tags "$(BUILDTAGS)"
 TMPDIR := /tmp/isula_build_tmpdir
 BEFLAG := -tmpdir=${TMPDIR}
-SAFEBUILDFLAGS := -buildid=IdByIsula -buildmode=pie -extldflags=-static -extldflags=-zrelro -extldflags=-znow $(LDFLAGS) $(BEFLAG)
+SAFEBUILDFLAGS := -buildid=IdByIsula -buildmode=pie -extldflags=-ftrapv -extldflags=-static -extldflags=-zrelro -extldflags=-znow $(LDFLAGS) $(BEFLAG)
 
 IMAGE_BUILDARGS := $(if $(http_proxy), --build-arg http_proxy=$(http_proxy))
 IMAGE_BUILDARGS += $(if $(https_proxy), --build-arg https_proxy=$(https_proxy))
-- 
2.27.0

