diff -ur isula-build-v0.9.6/Makefile isula-build-v0.9.6/Makefile
--- isula-build-v0.9.6/Makefile	2023-07-07 09:29:08.655822638 +0800
+++ isula-build-v0.9.6/Makefile	2023-07-07 09:30:26.783988012 +0800
@@ -22,8 +22,8 @@
 BUILDFLAGS := -tags "$(BUILDTAGS)"
 TMPDIR := /tmp/isula_build_tmpdir
 BEFLAG := -tmpdir=${TMPDIR}
-SAFEBUILDFLAGS := -buildid=IdByIsula -buildmode=pie -extldflags=-ftrapv -extldflags=-zrelro -extldflags=-znow $(BEFLAG) $(LDFLAGS)
-STATIC_LDFLAGS := -linkmode=external -extldflags "-static-pie -Wl,-z,now"
+SAFEBUILDFLAGS := -buildid=IdByIsula -buildmode=pie -extldflags=-ftrapv -extldflags=-zrelro -extldflags= $(BEFLAG) $(LDFLAGS)
+STATIC_LDFLAGS := -linkmode=external -extldflags "-Wl,-z,now"
 
 IMAGE_BUILDARGS := $(if $(http_proxy), --build-arg http_proxy=$(http_proxy))
 IMAGE_BUILDARGS += $(if $(https_proxy), --build-arg https_proxy=$(https_proxy))
